doc/developer/style.md

```diff
@@ -116,6 +116,77 @@ Some key principles to highlight are:
 * Avoid contractions and spell out words in full.
 * Word choices to be mindful of: find vs. exists, may vs. can vs. might.
 
+## Log message style
+
+We use the [`tracing` crate](https://docs.rs/tracing/latest/tracing/)'s log
+macros. Observe the following guidelines:
+
+  * Use sentence fragment style. Say "beginning operation", not
+    "Beginning operation."
+
+  * Prefer structured fields to string formatting:
```

Comment:
why does this matter? I personally find the flexibility of formatting a string however you want to be nice. does any of our tooling take advantage of the structure?